### PR TITLE
Fixes #2466 and #2465

### LINKED ISSUE #2466: Deserialization failures crash Java wrapper
**Description:**

If the deserialization throws a MolPicklerException, it isn't translated and something happens that causes the JVM to go down.

This is VERY BAD (TM)

### PR DESCRIPTION
**Pull request by greglandrum**
Two small fixes for substantial problems.
I fixed them together since they both came out of the same bug report

### CODE REVIEW

File: Code/JavaWrappers/gmwrapper/src-test/org/RDKit/PicklingTests.java [hunk 1]
```diff
@@ -134,6 +134,25 @@ public void testIssue220() {
...
 
+	@Test
+	public void testDepicklerCrash() {
+		// this was github #2466
+		String smi = "CC";
+		ROMol m1 = RWMol.MolFromSmiles(smi);
+		m1.getAtomWithIdx(0).setAtomMapNum(-1,false);
+		m1.getAtomWithIdx(1).setAtomMapNum(-1,false);
+		
+		Int_Vect pickle = m1.ToBinary();
```
Comment (bp-kelley) [inline Code/JavaWrappers/gmwrapper/src-test/org/RDKit/PicklingTests.java]: I’m a bit confused.  We shouldn’t be able to make a pickle that can’t be unpickled.

Comment (greglandrum) [reply Code/JavaWrappers/gmwrapper/src-test/org/RDKit/PicklingTests.java]: ah, good point. I was using this test before I fixed the bug that caused the crash.
I'll pull this test back out.

File: Code/GraphMol/MolPickler.cpp [hunk 2]
```diff
@@ -1321,7 +1321,7 @@ void MolPickler::_pickleAtom(std::ostream &ss, const Atom *atom) {
     streamWrite(ss, ENDQUERY);
   }
   if (getAtomMapNumber(atom, tmpInt)) {
-    if (tmpInt < 128) {
+    if (tmpInt >= 0 && tmpInt < 128) {
       tmpChar = static_cast<char>(tmpInt % 128);
```
Comment (bp-kelley) [inline Code/GraphMol/MolPickler.cpp]: Isn't the % 128 redundant here?

Comment (greglandrum) [reply Code/GraphMol/MolPickler.cpp]: Yep

**Pull request merged**